src/join_variant_summary_with_clinvar_alleles.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+import sys
+import pandas as pd
+
+from parse_clinvar_xml import HEADER
+
+FINAL_HEADER = HEADER + ['gold_stars', 'pathogenic', 'benign', 'conflicted']
+
+
+def join_variant_summary_with_clinvar_alleles(
+        variant_summary_table, clinvar_alleles_table,
+        genome_build_id="GRCh37"):
+    variant_summary = pd.read_csv(variant_summary_table, sep="\t",
+                                  index_col=False, compression="gzip")
+    print "variant_summary raw", variant_summary.shape
+
+    clinvar_alleles = pd.read_csv(clinvar_alleles_table, sep="\t",
+                                  index_col=False, compression="gzip")
+    print "clinvar_alleles raw", clinvar_alleles.shape
+
+    # use lowercase names and replace . with _ in column names:
+    variant_summary = variant_summary.rename(columns=dict(
+        (col, col.lower().replace(".", "_"))
+        for col in variant_summary.columns))
+    # rename first column to allele_id:
+    variant_summary = variant_summary.rename(
+        columns={variant_summary.columns[0]: "allele_id"})
+
+    # extract relevant columns for the correct assembly and
+    # rename clinicalsignificance, reviewstatus:
+    variant_summary = variant_summary[
+        variant_summary['assembly'] == genome_build_id]
+    variant_summary = variant_summary[
+        ['allele_id', 'clinicalsignificance', 'reviewstatus']]
+    variant_summary = variant_summary.rename(
+        columns={'clinicalsignificance': 'clinical_significance',
+                 'reviewstatus': 'review_status'})
+    print "variant_summary after filter", variant_summary.shape
+
+    # remove clinical_significance and review_status from clinvar_alleles:
+    clinvar_alleles = clinvar_alleles.drop(
+        ['clinical_significance', 'review_status'], axis=1)
+
+    # remove rows with multiple allele_ids
```

Comment:
In https://github.com/macarthur-lab/clinvar/blob/master/src/group_by_allele.py#L60-L65 `allele_id` is one of the columns that will be joined by `;` *if* there are more values, and the alleles are technically joined on chrom/pos/alt/ref: https://github.com/macarthur-lab/clinvar/blob/master/src/group_by_allele.py#L29.

So I have seen it in the wild! E.g. `227905;18626` popped up from a March dataset, so I added this.

I believe with the R script this was silently discarded because it just didn't match anything in `variant_summary` -- pandas is a bit more strict on datatypes, causing trouble.